backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -798,15 +807,43 @@ export class ExperimentService {
           partition.experiment = experimentDoc;
           return partition;
         });
-
+      
+      // creating queries docs
+      const promiseArray = [];
+      let queryDocsToSave =
+        (queries &&
+          queries.length > 0 &&
+          queries.map((query: Query) => {
+            promiseArray.push(this.metricRepository.findOne(query.metric));
+            // tslint:disable-next-line:no-shadowed-variable
+            const { createdAt, updatedAt, versionNumber, metric, ...rest } = query;
+            rest.experiment = experimentDoc;
+            rest.id = rest.id || uuid();
+            return rest;
+          })) ||
+        [];
+      if (promiseArray.length) {
+        const metricsDocs = await Promise.all([...promiseArray]);
+        queryDocsToSave = queryDocsToSave.map((queryDoc, index) => {
+          queryDoc.query.metric = metricsDocs[index];
```

Comment:
You should take care of scenario where metric doc doesn't exist

Reply:
ok